docs/integration_tests.md

```diff
@@ -0,0 +1,103 @@
+# Running integration tests manually
+
+The integration tests ensure HADDOCK3 workflows work properly by
+performing short docking simulations on benchmark systems. With these
+tests, we evaluate if the final scoring values in the developing branch
+are equal to those obtained in the `main` branch.
+
+You can run the integration tests manually using our helper scripts and
+following the guidelines here.
+
+You should run the integration tests every time you propose a new pull
+request.
+
+## 1. How to properly setup and run the integration tests?
+
+The easiest way is to have two installations of haddock3:
+one fixed at the `main` branch and other with your developments.
+
+Ensure the "HADDOCK3 main" repository and your HADDOCK3
+development repository are placed in the same parent folder. For
+example:
+
+```
+some-folder/
+    - haddock3/
+    - haddock3main/
+```
+
+## 2. Install HADDOCK3 main branch in a separate folder and python environment
+
+```bash
+git clone --recursive https://github.com/haddocking/haddock3 haddock3main
+cd haddock3main
+
+# rename the conda env name in requirements.yml file
+sed -i 's/haddock3/haddock3main/g' requirements.yml
+```
+
+Proceed installing the HADDOCK3 main following the installation
+instructions in the repository.
+
+This repository and environment will be the one you will use to run the
+test examples from the main branch.
+
+Once installed you don't need to install this "main" repository again.
+
+## 3. Running the tests in the "main"
+
+Before running the integration tests, ensure the "main" branch
+repository is up-to-date.
+
+```bash
+git pull
+```
+
+Navigate to the `examples/` folder and execute the tests:
+
+```bash
+python run_tests.py -b
+```
+
+This script will run all the test cases in the `examples/` directory and
+stop imminently if any of the tests break. For help, see `python
+run_tests.py -h`.
+
+If you are on a Slurm supported system, you can use the `test.job` file.
+You might need to edit it to match the paths and environment names.
+
+## 4. Running the tests on your branch
+
+Navigate to your development repository and haddock3 environment - you
+are likely on a different terminal window.
+
+Run the tests the same way:
+
+Navigate to the `examples/` folder and execute the tests:
+
+```bash
+python run_tests.py -b
+```
+
+If you are on a Slurm supported system, you can use the `run-tests.job`
+file.  You might need to edit the file according to your system
+specifications and to update paths and environment names.
+
+## 5. Comparing both runs
+
+Once both runs completed, that is, all `*-test.cfg` files were executed,
+you can compare the different results for the CAPRI scoring modules.
```

Comment:
Sorry, now I understand: you make a branch, make modifications, compare yours with the main and check the diff, ok.

But remember that due to the non-deterministic nature of haddock, changes in the CNS level might produce models with slight variations and these will be propagated to the capri metrics (ex 0.147 instead of 0.152). This is what I meant about values being within a certain range.

Reply:
Yes. I actually suggest keeping two separate haddock3 installations: one for development and the other for the `main`, just for this testing purposes. That's how I am doing.

Regarding the numbers, when running on the same machine, there should be not these different numbers, I think. So far, numbers have been always the same in all the PR tests I have made in the last months.

Obviously, if the changes affect the CAPRI scores on purpose, we cannot use these as validity metrics.

I will have a look further at your comments and edit accordingly.

Reply:
> Regarding the numbers, when running on the same machine, there should be not these different numbers, I think. 

You are wrong. I'm speaking from experience, but fine - we can cross that bridge when the times comes to it.

Reply:
Even for 3 decimal numbers? Okay. I can make it to test `close` numbers, that's not a problem. I was planning to make in Python as you suggested.

Reply:
As I said already, its not numerical fluctuation, its a result of a **slightly different model**, but nevermind that, let's not worry with future problems hehe

Reply:
Is not a future problem, is understanding haddock's behavior. So far, I see deterministic results: that is, running the examples several times always gives the exact same numbers. I am taking note of your comments :+1: